test/tsickle_test.ts

```diff
@@ -105,6 +105,21 @@ describe('emitWithTsickle', () => {
     expect(jsSources['b.js']).toContain(`export { Foo } from './a';`);
   });
 
+  it('should not go into an infinite loop with a self-referential type', () => {
+    const tsSources = {
+      'a.ts': `export function f() : typeof f { return f; }`,
+    };
+
+    const jsSources = emitWithTsickle(
+        tsSources, {
+          preserveConstEnums: true,
+          module: ts.ModuleKind.ES2015,
+        },
+        {es5Mode: false, googmodule: false});
+
+      expect(jsSources['a.js']).toContain(`export function f() { return f; }`);
```

Comment:
I would expect this output to contain some Closure stuff in it (e.g. "exports.f = ..."), is it possible this isn't running the right emit?

Reply:
The full emit that I saw was:
```ts
/**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @return {?}
     */
    export function f() { return f; }
``

So it looks like it is producing the closure type annotations (which I has assumed would be enough to trigger the bug).  Do you think trying goog.module emit would be needed to trigger the bug?

Reply:
I suspect this:
https://github.com/angular/tsickle/blob/3276f179c3d52ab5a7d3f2d88f57e92797f2bd1b/test/tsickle_test.ts#L37

This is the flag that we flipped in google but forgot about in this test suite.  :(

I think we should drop the flag entirely, but try flipping it here for now to see if that unsticks you.

Reply:
Aha, that was it. Thanks Evan!